### ISSUE TITLE
Doesn't work on Mavericks

### ISSUE THREAD

**Issue by rongarret** [github/SoftU2F#7]
SLSIA.  Installed on Mavericks, tried it in Chrome, didn't work.  Same setup works for hardware tokens.

**Comment by btoews** [comment]
Thanks for opening this. Can you check a few things for me?

If you run `ps aux | grep -i softu2f` do you see `/Applications/SoftU2F.app/Contents/MacOS/SoftU2F` running?

If so, can you try stopping that launchd task and running Soft U2F directly from the command line?

```
$ launchctl unload ~/Library/LaunchAgents/com.github.SoftU2F.plist
$ /Applications/SoftU2F.app/Contents/MacOS/SoftU2F
```

If the app starts correctly, you should see output like

```
$ /Applications/SoftU2F.app/Contents/MacOS/SoftU2F
Starting U2FHID thread
U2FHID thread started
Starting softu2f async run loop.
```

Otherwise you might see an error message or a crash.

If the app seems to be running at this point, can you go to https://demo.yubico.com/u2f?tab=register in Chrome and try making a U2F registration? If this were to work, you'd see output on the console like

```
Bens-MacBook-Pro:SoftU2F mastahyeti$ /Applications/SoftU2F.app/Contents/MacOS/SoftU2F
Starting U2FHID thread
U2FHID thread started
Starting softu2f async run loop.
Received frame:
	CID: 0xffffffff
	TYPE: INIT
	CMD: 0x06
	BCNTH: 0x00
	BCNTL: 0x08
	DATA: c4 00 ee ca 6f 4e c5 a3 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00

Sending frame:
	CID: 0xffffffff
	TYPE: INIT
	CMD: 0x06
	BCNTH: 0x00
	BCNTL: 0x11
	DATA: c4 00 ee ca 6f 4e c5 a3 01 00 00 00 02 00 00 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00

...
```

Thanks again for your help in debugging this.

**Comment by rongarret** [comment]
> If you run ps aux | grep -i softu2f do you see /Applications/SoftU2F.app/Contents/MacOS/SoftU2F running?

No.

> If so, can you try stopping that launchd task and running Soft U2F directly from the command line?

➔ /Applications/SoftU2F.app/Contents/MacOS/SoftU2F 
Killed: 9

That's probably why it's not working :-)

> Thanks again for your help in debugging this.

My pleasure.  Thanks for making this available!

**Comment by btoews** [comment]
Hmm. The `Killed: 9` output isn't very informative. I can try to get a Mavericks environment setup where I can test this myself, but in the meantime can you check a few more things for me?

1. Does `/Library/Extensions/softu2f.kext` exist?
1. What is the output of `sudo kextstat -b com.github.SoftU2FDriver`?

**Comment by rongarret** [comment]
> Does /Library/Extensions/softu2f.kext exist?

Yes.

> What is the output of sudo kextstat -b com.github.SoftU2FDriver?

➔ sudo kextstat -b com.github.SoftU2FDriver
Index Refs Address            Size       Wired      Name (Version) <Linked Against>
  140    0 0xffffff7f82427000 0x5000     0x5000     com.github.SoftU2FDriver (1) <28 5 4 3>

**Comment by DarwinAwardWinner** [comment]
In El Capitan, I get the following trying to run it on the command line:

```
$ /Applications/SoftU2F.app/Contents/MacOS/SoftU2F

dyld: Symbol not found: _kSecKeyAlgorithmECDSASignatureMessageX962SHA256
  Referenced from: /Applications/SoftU2F.app/Contents/MacOS/SoftU2F (which was built for Mac OS X 10.12)
  Expected in: /System/Library/Frameworks/Security.framework/Versions/A/Security
 in /Applications/SoftU2F.app/Contents/MacOS/SoftU2F
[1]    43009 trace trap  /Applications/SoftU2F.app/Contents/MacOS/SoftU2F
$ sudo kextstat -b com.github.SoftU2FDriver
Password:
Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
  155    0 0xffffff7f832d1000 0x5000     0x5000     com.github.SoftU2FDriver (1) C7E05742-7861-3AA3-866A-AD7FFA6317D5 <34 5 4 3>
```

**Comment by btoews** [comment]
Yep. That's going to be an issue. It looks like the [keychain features](https://developer.apple.com/documentation/security/seckeyalgorithm/1644015-ecdsasignaturemessagex962sha256) Soft U2F uses are only available on macOS 10.12+. I'll try to document this and I'll see if I can make the installer give a friendly message if it runs on an older OS. Thanks for the help debugging this.
